zeebe/zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/SnapshotOnlyDb.java

```diff
@@ -135,7 +135,7 @@ public void copySnapshot(
                   new DbNullKey(),
                   (prefixKey, prefixLength) -> {
                     try (final RocksIterator iterator = db.newIterator(readOptions)) {
-                      RawTransactionalColumnFamily.forEach(
+                      RawTransactionalColumnFamily.forEachPreallocated(
```

Comment:
:question: Did you make this change based on a performance test / profiling? If so, can you share the results?